src/main/java/com/avanza/gs/test/PartitionedPu.java

```diff
@@ -138,7 +136,7 @@ public boolean autostart() {
 	
 	@Override
 	public GigaSpace getClusteredGigaSpace() {
-		return GigaSpace.class.cast(getPrimaryInstanceApplicationContext(0).getBean(this.gigaSpaceBeanName)).getClustered();
+		return getPrimaryInstanceApplicationContext(0).getBean(GigaSpace.class, this.gigaSpaceBeanName).getClustered();
```

Comment:
The param order should probably be `name, Class`

Reply:
You are right, good catch!